hub/core/lock.py

```diff
@@ -0,0 +1,93 @@
+from typing import Tuple, Dict
+from hub.util.exceptions import LockedException
+from hub.util.path import get_path_from_storage
+from hub.util.threading import terminate_thread
+from hub.core.storage.provider import StorageProvider
+import hub
+import time
+import struct
+import atexit
+import uuid
+import os
+import threading
+import atexit
+
+
+class Lock(object):
+    def __init__(self, storage: StorageProvider):
+        self.storage = storage
+        self.acquired = False
+        self._thread_lock = threading.Lock()
+        self.acquire()
+        atexit.register(self.release)
```

Comment:
so, does this mean that if my script crashes, this dataset is unreadable forever?

Reply:
No, it will be readable after 5 mins (`hub.constants.DATASET_LOCK_VALIDITY`).